### PR TITLE
[#D3D-2731] Uses Mesh Normal for shadow bias

### PR DESCRIPTION
**Pull request by Kuranes**
ended up using normal for normal map to compute slope for bias, now uses specifically
Normal from mesh, and use it to do exploding offset normal shadow acne
mitigation

### CODE REVIEW

**Comment by cedricpinson** [comment]
dead end ?

**Comment by Kuranes** [comment]
it works, but need time to test on lots of models as it's rather touchy change.

File: sources/osgShader/node/lights.glsl [hunk 1]
```diff
@@ -76,65 +71,64 @@ vec3 computeSpotLightShading(
     // compute attenuation
     float attenuation = getLightAttenuation(dist, lightAttenuation);
     if (attenuation != 0.0)
+    {
```
Comment (cedricpinson) [inline sources/osgShader/node/lights.glsl]: coding style it should be if (...) { on the same line

**Comment by cedricpinson** [comment]
It would be good to first have a commit that remove all old shadow method that are not used. I am talking about esm ....

**Comment by Kuranes** [comment]
I started in another PR for that after this one, really easier for me that way.
( other one ia quite big, with  the light glsl style fix, conversion to Example template, etc. )

the clean PR raise two three question too, rhat might take some 'talkloop' removing other algo easy, but not sure waht you want to keep on the pcf algos (pcf, fakepcf, rotated, poisson, max number of taps?)

**Comment by Kuranes** [comment]
moved over https://github.com/cedricpinson/osgjs/pull/719 with everything bleached

**Pull request closed without merging**